core/internal/cltest/cltest.go

```diff
@@ -209,16 +209,20 @@ func NewEthConfirmer(t testing.TB, db *sqlx.DB, ethClient eth.Client, config evm
 type TestApplication struct {
 	t testing.TB
 	*chainlink.ChainlinkApplication
-	Logger   logger.Logger
-	Server   *httptest.Server
-	wsServer *httptest.Server
-	Started  bool
-	Backend  *backends.SimulatedBackend
-	Key      ethkey.KeyV2
+	Logger  logger.Logger
+	Server  *httptest.Server
+	Started bool
+	Backend *backends.SimulatedBackend
+	Key     ethkey.KeyV2
 }
 
-// NewWSServer returns a  new wsserver
-func NewWSServer(t *testing.T, msg string, callback func(data []byte)) (*httptest.Server, string) {
+// jsonrpcHandler is called with the method and request param(s).
+// respResult will be sent immediately. notifyResult is optional, and sent after a short delay.
+type jsonrpcHandler func(reqMethod string, reqParams gjson.Result) (respResult, notifyResult string)
+
+// NewWSServer starts a websocket server which invokes callback for each message received.
+// If chainID is set, then eth_chainId calls will be automatically handled.
+func NewWSServer(t *testing.T, chainID *big.Int, callback jsonrpcHandler) string {
```

Comment:
Adjusted to be more dynamic and to handle `eth_chainId` automatically when `chainID` is provided.